pkg/core/native/designate.go

```diff
@@ -120,14 +121,37 @@ func newDesignate(p2pSigExtensionsEnabled bool) *Designate {
 	return s
 }
 
-// Initialize initializes Oracle contract.
+// Initialize initializes Designation contract. It is called once at native Management's OnPersist
+// at the genesis block, and we can't properly fill the cache at this point, as there are no roles
+// data in the storage.
 func (s *Designate) Initialize(ic *interop.Context) error {
-	cache := &DesignationCache{}
-	cache.rolesChangedFlag = true
+	cache := &DesignationCache{
+		rolesChangedFlag: false,
```

Comment:
It was, but now it should be `false` in `Initialize`, because it is called for genesis only, no notifications are needed. And it should also be `true` in `InitializeCache` to notify services on node restart from non-empty database.